com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/resources/jsweb/dmloader.js

```diff
@@ -289,22 +347,31 @@ var GameArchiveLoader = {
         FileLoader.load(
             this._archiveLocationFilter(descriptionUrl),
             "json",
-            undefined,
-            function (loaded, total) { },
+            function (delta) { },
             function (error) { GameArchiveLoader.notifyFileDownloadError(descriptionUrl); },
-            function (json) { GameArchiveLoader.onReceiveDescription(json); });
+            function (json) { GameArchiveLoader.onReceiveDescription(json); },
+            function (loadedDelta, currentAttempt) { });
     },
 
     onReceiveDescription: function(json) {
+        var totalSize = json.total_size;
+        var exeName = CUSTOM_PARAMETERS['exe_name'];
         this._files = json.content;
-        this._totalDownloadBytes = 0;
-        this._currentDownloadBytes = 0;
 
-        // calculate total download size of all files
-        for(var i=0; i<this._files.length; ++i) {
-            this._totalDownloadBytes += this._files[i].size;
+        var isWASMSupported = Module['isWASMSupported'];
```

Comment:
Main change:
Now it starts loading game data and game engine concurrently after receiving `archive_files.json`. Here it calculates total amount of bytes that need to be downloaded.